### ISSUE TITLE
no validation when training

### ISSUE THREAD

**Issue by feimadada** [tianweiy/CenterPoint#279]
is there validation when training waymo open dataset?

**Comment by feimadada** [comment]
I want to print val loss after every epoch to monitor the training process

**Comment by tianweiy** [comment]
This is not supported now. Basically, Waymo validation will take ~1hr to run + ~0.5hr (or more than that) to evaluate on the validation set, which I feel is too long as a middle step between each training epoch. If you want to monitor the process, you can probably start a new single gpu job on the side to run dist_test.py and evaluate.   

You can also train for shorter iteration / subsample dataset for faster iteration and only evaluate the last epoch

**Comment by feimadada** [comment]
there is not evaluation for waymo in this repo, should I submit the prediction file to waymo official ?
![image](https://user-images.githubusercontent.com/39644765/157382217-cc257e98-364b-4901-ba47-0ff6d5e19d9e.png)

In fact, i just want to use the val loss to monitor the training process, not the mAP metric

**Comment by feimadada** [comment]
> prediction
I could downsample the val set to reduce the tine cost. how can i add some code to get the val loss? thansks a lot from a new pointcloud object detection learner

**Comment by feimadada** [comment]
i am training custom dataset ,I can't use waymo official metric

**Comment by tianweiy** [comment]
you can add some code here https://github.com/tianweiy/CenterPoint/blob/3c64610e78756990c8f67732749806a7a83caa5a/det3d/torchie/trainer/trainer.py#L423

**Comment by tianweiy** [comment]
and specify you want validation every x epoch at  ('val': x) https://github.com/tianweiy/CenterPoint/blob/3c64610e78756990c8f67732749806a7a83caa5a/configs/waymo/voxelnet/waymo_centerpoint_voxelnet_1x.py#L210
